inspirehep/modules/workflows/tasks/submission.py

```diff
@@ -330,21 +332,28 @@ def filter_keywords(obj, eng):
 @with_debug_logging
 def prepare_keywords(obj, eng):
     """Prepares the keywords in the correct format to be sent"""
-    prediction = obj.extra_data.get('keywords_prediction', {})
-    if not prediction:
-        return
-
+    prediction_keywords = get_value(obj.extra_data, 'keywords_prediction.keywords', [])
     keywords = obj.data.get('keywords', [])
-    for keyword in prediction.get('keywords', []):
+    for keyword in prediction_keywords:
         # TODO: differentiate between curated and guessed keywords
         keywords.append(
             {
                 'value': keyword['label'],
                 'source': 'curator' if keyword.get('curated') else 'magpie',
             }
```

Comment:
this is not actually ending up in the record, so either this task doesn't get called or there's some removal at a later stage.
In any case, we don't want these predicted keywords to end up in the record, but only those in this issue, and you should make sure they are really part of the record at the end.

Reply:
Discussed IRL.